core/src/unprocessed_packet_batches.rs

```diff
@@ -91,6 +102,42 @@ impl UnprocessedPacketBatches {
         UnprocessedPacketBatches(VecDeque::with_capacity(capacity))
     }
 
+    /// Insert new `deserizlized_packet_batch` into inner `VecDeque<DeserializedPacketBatch>`,
+    /// If buffer is at Max limit, packets (not packet_bacth) will be dropped, starting from
+    /// the lowest priority until an empty batch is located, then swap it with new batch;
+    /// Otherwise, new_batch will be pushed into the end of VecDeque;
+    /// returns tuple of (number of batch dropped, number of packets dropped)
+    pub fn insert_or_swap_batch(
```

Comment:
nit: maybe rename to `insert_by_priority`?

Reply:
Mmm, think I see your point. But when buffer is not full, it simply insert to end of it without needing priority, only when buffer is full and it has to replace a batch, it uses priority to find candidate. So maybe `insert_or_replay_by_priority`?

Reply:
hehe I'm beginning to think just calling it `insert` is fine, since it's the main entrypoint to `UnprocessedPacketBatches` anyways. Comment there is fine for explaining the inner workings

Reply:
Fair, changed to `insert_batch`